test/e2e_node/node_container_manager_test.go

```diff
@@ -56,9 +60,14 @@ func setDesiredConfiguration(initialConfig *kubeletconfig.KubeletConfiguration)
 
 var _ = framework.KubeDescribe("Node Container Manager [Serial]", func() {
 	f := framework.NewDefaultFramework("node-container-manager")
-	Describe("Validate Node Allocatable [NodeFeature:NodeAllocatable]", func() {
-		It("set's up the node and runs the test", func() {
-			framework.ExpectNoError(runTest(f))
+	Context(fmt.Sprintf("with node PID limits enabled"), func() {
+		tempSetCurrentKubeletConfig(f, func(initialConfig *kubeletconfig.KubeletConfiguration) {
```

Comment:
I would prefer just setting the kubelet config in one place, so we don't have to wait for the kubelet to restart multiple times before running the test.  Can we move setting the feature gate to the `setDesiredConfiguration` function at the top?

Alternatively, you could modify the test to use `tempSetCurrentKubeletConfig` for all configuration modification, but that isn't a requirement for this PR.

Reply:
Sure, done.  It is indeed faster this way.